### PR TITLE
Fix snapshotting validation and rewrite -s to --rebuild - Closes #2843, #3119

### LINKED ISSUE #2843: Only accept integer range values for snapshotting rounds
### Expected behavior
We should accept only the integer range values for snapshotting rounds. If need to specify the full all available rounds we should use `-1` for it. 

To run full rounds snapshot we should be able to do;

```
node app.js -s -1
```

### Actual behavior
Currently we are providing different values, integer for valid rounds and non-integer for full rounds. 
https://github.com/LiskHQ/lisk/blob/cb68ebb96aacc3dccd532de6f926a1c39a2b1411/modules/loader.js#L772

e.g. Currently we can run something like this 

```
node app.js -s all
or
node app.js -s highest
or 
node app.js -s my-name
```

### Which version(s) does this affect? (Environment, OS, etc...)
all

### PR DESCRIPTION
**Pull request by michielmulders**
### How did I fix it?
- Change parameter for snapshotting from `-s` to `-b` (`--rebuild`).
- Remove all occurrences of `snapshot` and rename to `rebuild`.
- Modify validation logic to accept `0` as input for `--rebuild`.

### How to test it?
Run all the tests.

### Review checklist

* The PR resolves #2843, #3119 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Comment by michielmulders** [comment]
Green PR two days ago: https://jenkins.lisk.io/blue/organizations/jenkins/lisk-sdk-framework/detail/PR-3420/2/pipeline

Fix merge conflicts: running checks again.

File: lisk/build/target/etc/rebuild.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
 	"httpPort": 9000,
 	"wsPort": 9001,
-	"logFileName": "logs/lisk_snapshot.log",
+	"logFileName": "logs/lisk_rebuild.log",
 	"fileLogLevel": "info",
 	"db": {
-		"database": "lisk_snapshot"
+		"database": "lisk_rebuild"
```
Comment (nazarhussain) [inline lisk/build/target/etc/rebuild.json]: @michielmulders That change required to align changes in scripts as well. Please discuss with @fchavant and create respective issue for it.

Comment (michielmulders) [reply lisk/build/target/etc/rebuild.json]: Aligned with @fchavant , `lisk/build/target/etc/rebuild.json` and `lisk/build/target/etc/pm2-rebuild.json` can be safely removed. No further influence. Thanks!

**Comment by fchavant** [comment]
@michielmulders please do not remove `lisk_snapshot.sh`
@nazarhussain since https://github.com/LiskHQ/lisk-sdk/pull/3114 `lisk_snapshot.sh` does not perform (what we now call) a rebuild; thus it should not be removed or renamed.

**Comment by nazarhussain** [comment]
@fchavant I suggest not to remove or change the file content, I checked the current file and saw its just taking backup of the database. Where is that functionality of rebuilding is now used in scripts?

**Comment by fchavant** [comment]
@nazarhussain `lisk_snapshot.sh` does *not* rebuild; see https://github.com/LiskHQ/lisk-sdk/issues/3109 and https://github.com/LiskHQ/lisk-sdk/pull/3114

**Comment by nazarhussain** [comment]
@fchavant Yes I checked it does not rebuild, I am asking the rebuild functionality which this PR is refactoring, where we are using it now in the scripts?

**Comment by fchavant** [comment]
@nazarhussain we are not.

**Pull request merged**